### PR TITLE
[vcpkg] add updated compile wrapper for clang-cl

### PR DESCRIPTION
**Pull request by Neumann-A**
has already been merged upstream but I have no clue when it will be in automake from msys.
The relevant change is line 259. 

Alternatively we could just patch the msys download? But where should those patches be stored?

### CODE REVIEW

**Comment by JavierMatosD** [comment]
> has already been merged upstream but I have no clue when it will be in automake from msys. The relevant change is line 259.
> 
> Alternatively we could just patch the msys download? But where should those patches be stored?

I am concerned about the potential licensing implications of this. Is it unreasonable to wait for msys?

**Comment by Neumann-A** [comment]
> I am concerned about the potential licensing implications of this. Is it unreasonable to wait for msys?

I feared that that would be the answer. So where should I store the patch for msys ? ;)

**Comment by ras0219-msft** [comment]
`scripts/msys/xyz.patch`; `vcpkg_acquire_msys()` then can apply the patch after un-tarring all the packages together but before renaming if it detects that the automake package was requested.

File: scripts/cmake/vcpkg_acquire_msys.cmake [hunk 1]
```diff
@@ -68,9 +68,16 @@ function(z_vcpkg_acquire_msys_declare_package)
...
+        list(APPEND Z_VCPKG_MSYS_ARCHIVES "${arg_NAME}")
         set(Z_VCPKG_MSYS_ARCHIVES "${Z_VCPKG_MSYS_ARCHIVES}" PARENT_SCOPE)
+        set(Z_VCPKG_MSYS_${arg_NAME}_ARCHIVE "${archive}" PARENT_SCOPE)
+        set(Z_VCPKG_MSYS_${arg_NAME}_PATCHES "${arg_PATCHES}" PARENT_SCOPE)
         string(APPEND Z_VCPKG_MSYS_TOTAL_HASH "${arg_SHA512}")
+        foreach(patch IN LISTS arg_PATCHES)
+            file(SHA512 "${patch}" patch_sha)
+            string(APPEND Z_VCPKG_MSYS_TOTAL_HASH "${patch_sha}")
+        endforeach()
+        set(Z_VCPKG_MSYS_TOTAL_HASH "${Z_VCPKG_MSYS_TOTAL_HASH}" PARENT_SCOPE)
```
Comment (Neumann-A) [inline scripts/cmake/vcpkg_acquire_msys.cmake:80]:
```suggestion
```

**Comment by Neumann-A** [comment]
@ras0219-msft: Done.

**Comment by JavierMatosD** [comment]
Thanks!

**Comment by Neumann-A** [comment]
@dg0yt Couldn't you also patch libtool with this? Or am I pinging the wrong person.

**Comment by dg0yt** [comment]
@Neumann-A libtool is considered external and not accepted as a port.

**Comment by Neumann-A** [comment]
> @Neumann-A libtool is considered external and not accepted as a port.

didn't you have changes to bash scripts which could now be patched into `vcpkg_acquire_msys` ?

**Comment by dg0yt** [comment]
@Neumann-A Sorry, the PR title was unable to convey the information that this PR contains relevant changes, such as a `PATCHES` option.

**Pull request merged**